Marlin/Marlin.h

```diff
@@ -365,4 +372,13 @@ extern uint8_t active_extruder;
 
 void calculate_volumetric_multipliers();
 
+// Buzzer
+#if HAS_BUZZER
+  #if ENABLED(SPEAKER)
+    extern Speaker buzzer;
```

Comment:
```cpp
In file included from sketch\Marlin.h:72:0,

                 from sketch\M100_Free_Mem_Chk.cpp:43:

speaker.h:30: error: conflicting declaration 'Speaker buzzer'

 extern Speaker buzzer;

                ^

In file included from sketch\Marlin.h:70:0,

                 from sketch\M100_Free_Mem_Chk.cpp:43:

buzzer.h:33: error: 'buzzer' has a previous declaration as 'Buzzer buzzer'

 extern Buzzer buzzer;

               ^

exit status 1
conflicting declaration 'Speaker buzzer'
```

I am having issues due to inheritance, I could have on the base class a `"#if !ENABLED(SPEAKER)` but then the base class "knows" about the "top" class which breaks encapsulation also. Any ideas ?